code/mission/missionparse.cpp

```diff
@@ -2489,6 +2489,24 @@ void parse_bring_in_docked_wing(p_object *p_objp, int wingnum, int shipnum)
 	// make sure we haven't created too many ships
 	Assert(wingp->current_count <= MAX_SHIPS_PER_WING);
 
+	//fixes bug where docked wings are not called with the 'On Ship Arrive' hook variable --wookieejedi
+	int num_ships = wingp->current_count;
+	int anchor_objnum = -1;
+	anchor_objnum = mission_set_arrival_location(wingp->arrival_anchor, wingp->arrival_location, wingp->arrival_distance, OBJ_INDEX(leader_objp), wingp->arrival_path_mask, &pos, &orient);
```

Comment:
Does `mission_set_arrival_location` change something in the mission? I'm worried that this might cause changes in behaviour.

Also, you can initialize `anchor_objnum` directly from the return value of `mission_set_arrival_location` and you can move the initialization inside the `if` below.

Reply:
Ah you're right I think it does. I won't use that function then. In that function, anchor_objnum = Ships[shipnum].objnum, so perhaps I should use that instead of using the whole function.

Also, should I change NULL to nullptr in this case?

Reply:
Yes, you always need to fix those things now.